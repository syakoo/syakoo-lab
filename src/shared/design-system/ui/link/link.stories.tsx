import type { Meta } from "@storybook/nextjs";

import { theme } from "@/shared/design-system/theme.css";

import { Link } from ".";

const meta = {
  component: Link,
  tags: ["autodocs"],
  parameters: { testLevel: "snapshot" },
} satisfies Meta<typeof Link>;

export default meta;

export const Default = () => {
  return <Link href="#">Default Link</Link>;
};

export const Variants = () => {
  return (
    <>
      <div>
        <Link colored href="#">
          Colored Link
        </Link>
      </div>
      <div>
        <Link href="#" underlined>
          Underlined Link
        </Link>
      </div>
      <div>
        <Link href="#" noHovered>
          no hovered Link
        </Link>
      </div>
      <div>
        <Link href="#" noTransparent>
          no transparent Link
        </Link>
      </div>
      <div>
        <Link display="block" href="#">
          <div
            style={{
              border: `1px solid ${theme.color.brand.primary}`,
              borderRadius: theme.radius[100],
            }}
          >
            Block Link
          </div>
        </Link>
      </div>
    </>
  );
};
