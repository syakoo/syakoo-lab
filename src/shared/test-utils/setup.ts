import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("@mdx-js/mdx", () => {
  return {
    runSync: jest
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
jest.mock("mermaid", () => ({
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

jest.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
