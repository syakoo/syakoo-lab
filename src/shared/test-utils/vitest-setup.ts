import { vi } from "vitest";

import { exampleMDXComponent } from "@/features/mdx/mocks/fixture-mdxcomponent";

import { loadEnv } from "./nextjs/env";

// NOTE: Next.jsのImageコンポーネントをテスト用にモック
vi.mock("next/image", () => ({
  default: "img",
}));

// NOTE: `@mdx-js/mdx` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("@mdx-js/mdx", () => {
  return {
    runSync: vi
      .fn()
      .mockImplementation(() => ({ default: exampleMDXComponent })),
  };
});

// NOTE: `mermaid` の esm がうまく解決できなかったためまとめてモック化
// これが原因でおかしくなっていたらごめんなさい
vi.mock("mermaid", () => ({
  default: {
    initialize: () => {
      return;
    },
    run: () => {
      return;
    },
  },
  initialize: () => {
    return;
  },
  run: () => {
    return;
  },
}));

loadEnv();

vi.spyOn(Math, "random").mockImplementation(() => 0.5866245602626035);
