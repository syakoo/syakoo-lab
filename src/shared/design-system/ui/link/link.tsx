import type { RecipeVariants } from "@vanilla-extract/recipes";
import NextLink from "next/link";
import type React from "react";

import { linkStyle } from "./link.css";

type LinkProps = {
  href: string;
  children: React.ReactNode;
} & Partial<RecipeVariants<typeof linkStyle>> &
  Omit<React.ComponentPropsWithoutRef<"a">, "href" | "children">;

export const Link: React.FC<LinkProps> = ({
  href,
  children,
  colored = false,
  underlined = false,
  noHovered = false,
  noTransparent = false,
  display,
  ...otherProps
}) => {
  return (
    <NextLink
      className={`${linkStyle({
        colored,
        underlined,
        noHovered,
        noTransparent,
        display,
      })}`}
      href={href}
      {...otherProps}
    >
      {children}
    </NextLink>
  );
};
