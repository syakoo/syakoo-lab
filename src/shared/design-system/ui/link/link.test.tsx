import { testStories } from "@/shared/test-utils/test-stories";

import * as stories from "./link.stories";

describe("Link", () => {
  testStories(stories);
});
