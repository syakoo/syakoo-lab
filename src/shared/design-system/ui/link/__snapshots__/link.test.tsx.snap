// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link スナップショット Default 1`] = `
<body>
  <div>
    <a
      class="link_linkStyle__1vz8ftz0"
      href="#"
    >
      Default Link
    </a>
  </div>
</body>
`;

exports[`Link スナップショット Variants 1`] = `
<body>
  <div>
    <div>
      <a
        class="link_linkStyle__1vz8ftz0 link_linkStyle_colored_true__1vz8ftz1"
        href="#"
      >
        Colored Link
      </a>
    </div>
    <div>
      <a
        class="link_linkStyle__1vz8ftz0 link_linkStyle_underlined_true__1vz8ftz2"
        href="#"
      >
        Underlined Link
      </a>
    </div>
    <div>
      <a
        class="link_linkStyle__1vz8ftz0 link_linkStyle_noHovered_true__1vz8ftz3"
        href="#"
      >
        no hovered Link
      </a>
    </div>
    <div>
      <a
        class="link_linkStyle__1vz8ftz0 link_linkStyle_display_block__1vz8ftz4"
        href="#"
      >
        <div
          style="border-radius: var(--radius-100__1874wf71g);"
        >
          Block Link
        </div>
      </a>
    </div>
  </div>
</body>
`;
